packages/cdk-codepipeline-slack/src/slack-notifier.ts

```diff
@@ -0,0 +1,90 @@
+import * as path from 'path';
+import { Construct } from '@aws-cdk/core';
+import { Code, Function, Runtime } from '@aws-cdk/aws-lambda';
+import { Pipeline } from '@aws-cdk/aws-codepipeline';
+import { LambdaFunction } from '@aws-cdk/aws-events-targets';
+import { Rule } from '@aws-cdk/aws-events';
+import { PolicyStatement } from '@aws-cdk/aws-iam';
+
+export interface SlackNotifierProps {
+    slackBotToken: string;
+    slackSigningSecret: string;
+    slackChannel?: string;
+    slackChannelId?: string;
+    slackBotName?: string;
+    slackBotIcon?: string;
+    pipeline: Pipeline;
+    stageNames?: string[];
+}
+
+export class SlackNotifier extends Construct {
+    constructor(scope: Construct, id: string, props: SlackNotifierProps) {
+        super(scope, id);
+
+        const {
+            slackBotToken,
+            slackSigningSecret,
+            slackChannel,
+            slackChannelId,
+            slackBotName,
+            slackBotIcon,
+            pipeline,
+            stageNames,
+        } = props;
+
+        const environment: Record<string, string> = {
+            SLACK_BOT_TOKEN: slackBotToken,
+            SLACK_SIGNING_SECRET: slackSigningSecret,
+            SLACK_CHANNEL: slackChannel || '',
```

Comment:
Do we want to add a consistency check here to check if channelName or channelId has been set?

https://github.com/aws/aws-cdk/blob/32b98a8b0c85b3af656db8293b37a7551726515b/packages/%40aws-cdk/core/lib/construct.ts#L545

Reply:
consistency check added (both missing or duplicate config)